src/web/client/WebExtensionContext.ts

```diff
@@ -627,6 +661,98 @@ class WebExtensionContext implements IWebExtensionContext {
     public getVscodeWorkspaceState(key: string): IEntityInfo | undefined {
         return this._vscodeWorkspaceState.get(key);
     }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    public async getWorkerScript(workerUrl : URL) : Promise<any> {
+        try {
+            this.telemetry.sendInfoTelemetry(
+                telemetryEventNames.WEB_EXTENSION_FETCH_WORKER_SCRIPT
+            );
+
+            const response = await this.concurrencyHandler.handleRequest(
+                workerUrl
+            )
+
+            if (!response.ok) {
+                throw new Error(
+                    `Failed to fetch worker script '${workerUrl.toString()}': ${response.statusText}`
+                );
+            }
+
+            this.telemetry.sendInfoTelemetry(
+                telemetryEventNames.WEB_EXTENSION_FETCH_WORKER_SCRIPT_SUCCESS,
+                { workerUrl: workerUrl.toString() }
+            );
+
+            return await response.text();
+        } catch (error) {
+            this.telemetry.sendErrorTelemetry(
+                telemetryEventNames.WEB_EXTENSION_FETCH_WORKER_SCRIPT_FAILED,
+                this.getWorkerScript.name,
+                (error as Error)?.message,
```

Comment:
Please avoid using telemetry as replacement for logging. Why is there a need for sending entire error message to telemetry?

@ashishchoudhary001

Reply:
@petrochuk We have been using this pattern in the extension for a while now. This log ends up in the Application Insights cluster and not in Aggregate cluster. This helps us debug any failures in prod. Any better suggestions here?

Reply:
@BidishaMS can you confirm this part "This log ends up in the Application Insights cluster and not in Aggregate cluster."

Reply:
+1 to @tyaginidhi . We have been logging entire error log trace since a while. Moreover it is not in Aggregate cluster. It is only flown to PROD i.e. App Insights

Reply:
This code runs in VSCode Web. Here we don't have the option to log in PAC Log file.

Reply:
@petrochuk  can you please check comments above - do we need to re-review the telemetry code here still?